lib/AST/RawComment.cpp

```diff
@@ -154,7 +154,9 @@ RawComment Decl::getRawComment(bool SerializedOK) const {
         if (!CachedLocs->DocRanges.empty()) {
           SmallVector<SingleRawComment, 4> SRCs;
           for (const auto &Range : CachedLocs->DocRanges) {
-            SRCs.push_back({ Range, Context.SourceMgr });
+            if (Range.isValid()) {
+              SRCs.push_back({ Range, Context.SourceMgr });
+            }
```

Comment:
This might cause comments without ranges to not load at all, which I'm not sure is what you want. Can you use the other `SingleRawComment` constructor here in an `else` clause?

Reply:
I don't think there's a way to load the comment text from this block of code, as far as i can tell. If i understand it correctly, it seems like comments without ranges would skip this loop to begin with (since there would be no serialized locations in the Decl) and instead get loaded by the call to `Unit->getCommentForDecl(this)` below.

Reply:
Ah nevermind, you don't have the text available here. I wonder if you should clear `SRCs` and break out of the loop then since the other ranges are likely to be missing at this point. Anyway, looks reasonable.

Reply:
That sounds like a good idea, at least as an optimization to skip running over the list of ranges. I've pushed a new commit with this change.